### DISCUSSION TITLE
Client/Server or-tools

### DISCUSSION THREAD

**Discussion by samirsebbah** [google/or-tools#3598] [Feature requests]
Folks,
Just wondering if there are initiatives around separating the monolith or-tools lib into client/server modules. A Client to generate model instances (e.g., lp. xxx, proto)+search direction + solver + ... and a Server to solve those models. Can you please share your thoughts.

Thanks

**Comment by lperron** [comment]
Which solver ?

**Reply by samirsebbah** [reply]
any of the standard (I)LP or CSP based solvers. The payload would contain the model specification (e.g., lp or high level cp) and an indication of a solver to use (golp, cplex. cbc, Gurobi, etc) and a search direction, etc.
